packages/config/src/api/site_info.ts

```diff
@@ -37,30 +36,58 @@ export const getSiteInfo = async function ({
 }: GetSiteInfoOpts) {
   const { env: testEnv = false } = testOpts
 
-  if (api === undefined || mode === 'buildbot' || testEnv) {
+  if (api === undefined || testEnv || offline) {
     const siteInfo = siteId === undefined ? {} : { id: siteId }
 
-    const integrations = mode === 'buildbot' && !offline ? await getIntegrations({ siteId, testOpts, offline }) : []
-
-    return { siteInfo, accounts: [], addons: [], integrations }
+    return { siteInfo, accounts: [], addons: [], integrations: [] }
   }
 
-  const promises = [
-    getSite(api, siteId, siteFeatureFlagPrefix),
-    getAccounts(api),
-    getAddons(api, siteId),
-    getIntegrations({ siteId, testOpts, offline }),
-  ]
+  const siteInfo = await getSite(api, siteId, siteFeatureFlagPrefix)
+  const featureFlags = siteInfo.feature_flags
 
-  const [siteInfo, accounts, addons, integrations] = await Promise.all(promises)
+  const useV2Endpoint = featureFlags?.cli_integration_installations_meta
 
-  if (siteInfo.use_envelope) {
-    const envelope = await getEnvelope({ api, accountId: siteInfo.account_slug, siteId, context })
+  if (useV2Endpoint) {
+    const promises = [
+      getAccounts(api),
+      getAddons(api, siteId),
+      getIntegrations({ siteId, testOpts, offline, accountId: siteInfo.account_id, featureFlags }),
+    ]
 
-    siteInfo.build_settings.env = envelope
-  }
+    const [accounts, addons, integrations] = await Promise.all(promises)
+
+    if (siteInfo.use_envelope) {
+      const envelope = await getEnvelope({ api, accountId: siteInfo.account_slug, siteId, context })
+
+      siteInfo.build_settings.env = envelope
+    }
+
+    return { siteInfo, accounts, addons, integrations }
+  } else {
```

Comment:
what's about mode 'buildbot' shouldn't be this covered in this part as well?

Why is the mode buildbot only covered in the non feature flaged way?

Reply:
If I am honest, I don't know why we did this and I opted to keep this as it was previously written so that I could be sure I didn't regress the previous functionality as I am only interested in the adding the v2 behaviour